zvariant_derive/tests/tests.rs

```diff
@@ -28,15 +28,37 @@ fn derive_struct() {
 
 #[test]
 fn derive_enum() {
-    #[repr(u32)]
-    #[derive(Type)]
+    #[repr(u16)]
+    #[derive(Debug, Clone, Copy, PartialEq, Type, Value, OwnedValue)]
     enum RequestNameFlags {
         AllowReplacement = 0x01,
         ReplaceExisting = 0x02,
         DoNotQueue = 0x04,
     }
 
-    assert_eq!(RequestNameFlags::signature(), "u")
+    assert_eq!(RequestNameFlags::signature(), "q");
+    assert_eq!(Value::from(RequestNameFlags::AllowReplacement), Value::U16(0x01));
+    assert_eq!(RequestNameFlags::try_from(Value::U16(0x04)), Ok(RequestNameFlags::DoNotQueue));
+    assert_eq!(RequestNameFlags::try_from(Value::U16(0x03)), Err(Error::IncorrectType));
+}
+
+#[test]
+fn derive_enum_no_repr() {
+    #[derive(Debug, Clone, Copy, PartialEq, Type, Value, OwnedValue)]
```

Comment:
* same here: try and modify existing tests (preferably in `zvariant` as people often copy code from tests and I want to direct use of macro crates these days).
* Please only add what is needed. E.g do we need `Clone` and `Copy` here? :thinking: